### PR TITLE
HHH-13811 - Test and fix for H2 boolean literal issue

### PR DESCRIPTION
**Pull request by beikov**
https://hibernate.atlassian.net/browse/HHH-13811

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/dialect/H2Dialect.java [hunk 1]
```diff
@@ -423,6 +423,11 @@ public boolean requiresParensForTupleDistinctCounts() {
 		return true;
 	}
 
+	@Override
+	public String toBooleanValueString(boolean bool) {
+		return bool ? "true" : "false";
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/dialect/H2Dialect.java]: So I checked other dialect class implementing this method. It seems the following pattern is followed invariably:
```
return String.valueOf( bool );
```
so maybe we might do that as well?

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/dialect/H2Dialect.java]: I actually copied this from the PostgreSQL81Dialect(https://github.com/hibernate/hibernate-orm/blob/e896db2ce2cdcaf2ed08e7333343b79cc228dc59/hibernate-core/src/main/java/org/hibernate/dialect/PostgreSQL81Dialect.java#L410), so it's not consistent in the code base. I don't see any advantage of using one over the other to be honest.

Comment (NathanQingyangXu) [reply hibernate-core/src/main/java/org/hibernate/dialect/H2Dialect.java]: okay, if so, that is fine.

**Comment by dreab8** [comment]
Hi @beikov ,

it seems that the test does not fail even when I remove the `H2Dialect` impl of the `toBooleanValueString(boolean bool)` method.

**Comment by jwgmeligmeyling** [comment]
@beikov perhaps it depends on the H2 version used? Weve seen before that older versions of H2 are pretty limited sometimes

**Comment by beikov** [comment]
Forget the issue and the PR. The problem is that H2 simply doesn't support comparing a predicate like `EXISTS` with a _booleanish_ literal, so no matter what, you always have to use a case when expression.

**Pull request closed without merging**